packages/fetchai/skills/aries_alice/behaviours.py

```diff
@@ -0,0 +1,173 @@
+        self._admin_port = kwargs.pop("admin_port", DEFAULT_ADMIN_PORT)
+        self._admin_url = "http://{}:{}".format(self.admin_host, self.admin_port)
+
+        services_interval = kwargs.pop(
+            "services_interval", DEFAULT_SERVICES_INTERVAL
+        )  # type: int
+        super().__init__(tick_interval=services_interval, **kwargs)
+
+    @property
+    def admin_host(self) -> str:
+        """Get the admin host."""
+        return self._admin_host
+
+    @property
+    def admin_port(self) -> str:
+        """Get the admin port."""
+        return self._admin_port
+
+    @property
+    def admin_url(self) -> str:
+        """Get the admin URL."""
+        return self._admin_url
+
+    def setup(self) -> None:
+        """
+        Implement the setup.
+
+        :return: None
+        """
+        self.context.logger.info("My address is: " + self.context.agent_address)
+        self._register_agent()
+        self._register_service()
+
+    def act(self) -> None:
+        """
+        Implement the act.
+
+        :return: None
+        """
+
+    def teardown(self) -> None:
+        """
+        Implement the task teardown.
+
+        :return: None
+        """
+        self._unregister_service()
+        self._unregister_agent()
+
+    def _register_agent(self) -> None:
+        """
+        Register the agent's location.
+
+        :return: None
+        """
+        strategy = cast(AliceStrategy, self.context.strategy)
+        description = strategy.get_location_description()
+        oef_search_dialogues = cast(
+            OefSearchDialogues, self.context.oef_search_dialogues
+        )
+        oef_search_msg = OefSearchMessage(
+            performative=OefSearchMessage.Performative.REGISTER_SERVICE,
+            dialogue_reference=oef_search_dialogues.new_self_initiated_dialogue_reference(),
+            service_description=description,
+        )
+        oef_search_msg.counterparty = self.context.search_service_address
+        oef_search_dialogues.update(oef_search_msg)
+        self.context.outbox.put_message(message=oef_search_msg)
+        self.context.logger.info("registering Alice on SOEF.")
+
+    def _register_service(self) -> None:
+        """
+        Register the agent's service.
+
+        :return: None
+        """
+        strategy = cast(AliceStrategy, self.context.strategy)
+        description = strategy.get_register_service_description()
+        oef_search_dialogues = cast(
+            OefSearchDialogues, self.context.oef_search_dialogues
+        )
+        oef_search_msg = OefSearchMessage(
+            performative=OefSearchMessage.Performative.REGISTER_SERVICE,
+            dialogue_reference=oef_search_dialogues.new_self_initiated_dialogue_reference(),
+            service_description=description,
+        )
+        oef_search_msg.counterparty = self.context.search_service_address
+        oef_search_dialogues.update(oef_search_msg)
```

Comment:
You should always! check the return value of `oef_search_dialogues.update()`. If it is none you need to throw and assertion error, otherwise you will get weird errors.

Reply:
I will make the change.

I copied this from the generic buyer/seller skills. If this is the desired pattern, note that it is not implemented in the those skills.